File: frontend/src/types/entrepreneurship.ts
export enum VentureStatus {
  IDEA = 'idea',
  DEVELOPMENT = 'development',
  LAUNCHED = 'launched',
  FUNDED = 'funded',
  CLOSED = 'closed',
}

export enum CompetitionStatus {
  UPCOMING = 'upcoming',
  OPEN = 'open',
  JUDGING = 'judging',
  COMPLETED = 'completed',
  CANCELLED = 'cancelled',
}

export enum MentorshipStatus {
  PENDING = 'pending',
  ACTIVE = 'active',
  COMPLETED = 'completed',
  CANCELLED = 'cancelled',
}

export enum FundingStatus {
  REQUESTED = 'requested',
  UNDER_REVIEW = 'under_review',
  APPROVED = 'approved',
  DISBURSED = 'disbursed',
  REJECTED = 'rejected',
}

export interface StudentVenture {
  id: number;
  institution_id: number;
  venture_name: string;
  tagline?: string;
  founder_students: number[];
  primary_founder_id: number;
  business_idea: string;
  problem_statement: string;
  solution: string;
  target_market: string;
  revenue_model?: string;
  pitch_deck_url?: string;
  business_plan_url?: string;
  logo_url?: string;
  website_url?: string;
  mentor_id?: number;
  funding_requested?: number;
  funding_received: number;
  currency: string;
  venture_status: VentureStatus;
  metrics?: VentureMetrics;
  customers: number;
  revenue: number;
  social_impact?: string;
  milestones?: Milestone[];
  team_roles?: TeamRole[];
  achievements?: Achievement[];
  pitch_competition_participations?: number[];
  awards?: Award[];
  is_active: boolean;
  is_featured: boolean;
  created_at: string;
  updated_at: string;
  primary_founder?: StudentInfo;
  mentor?: EntrepreneurshipMentor;
  team_members?: StudentInfo[];
}

export interface VentureMetrics {
  customer_acquisition_rate?: number;
  monthly_revenue_growth?: number;
  social_impact_score?: number;
  user_retention_rate?: number;
  market_validation_score?: number;
}

export interface Milestone {
  id: string;
  title: string;
  description: string;
  target_date: string;
  completed: boolean;
  completed_date?: string;
}

export interface TeamRole {
  student_id: number;
  role: string;
  responsibilities: string[];
}

export interface Achievement {
  id: string;
  title: string;
  description: string;
  date: string;
  type: string;
}

export interface Award {
  id: string;
  name: string;
  organization: string;
  date: string;
  amount?: number;
}

export interface StudentInfo {
  id: number;
  first_name: string;
  last_name: string;
  grade: string;
  photo_url?: string;
}

export interface PitchCompetition {
  id: number;
  institution_id: number;
  competition_name: string;
  description: string;
  theme?: string;
  judges?: number[];
  judge_details?: JudgeInfo[];
  prize_pool?: number;
  currency: string;
  prizes?: Prize[];
  submission_deadline: string;
  competition_date?: string;
  eligibility_criteria?: EligibilityCriteria;
  evaluation_criteria?: EvaluationCriteria[];
  max_participants?: number;
  current_participants: number;
  status: CompetitionStatus;
  winner_venture_id?: number;
  runners_up?: number[];
  leaderboard?: LeaderboardEntry[];
  final_results?: FinalResult[];
  recording_url?: string;
  highlights_url?: string;
  is_public: boolean;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface JudgeInfo {
  id: number;
  name: string;
  title: string;
  organization: string;
  photo_url?: string;
}

export interface Prize {
  rank: number;
  title: string;
  amount?: number;
  description?: string;
}

export interface EligibilityCriteria {
  min_grade?: string;
  max_grade?: string;
  venture_status?: VentureStatus[];
  max_team_size?: number;
}

export interface EvaluationCriteria {
  id: string;
  name: string;
  description: string;
  max_score: number;
  weight: number;
}

export interface LeaderboardEntry {
  venture_id: number;
  venture_name: string;
  total_score: number;
  rank: number;
}

export interface FinalResult {
  venture_id: number;
  venture_name: string;
  rank: number;
  total_score: number;
  prize?: string;
}

export interface PitchSubmission {
  id: number;
  institution_id: number;
  competition_id: number;
  venture_id: number;
  pitch_video_url?: string;
  presentation_url?: string;
  supporting_documents?: SupportingDocument[];
  submission_date: string;
  judge_scores?: JudgeScore[];
  total_score?: number;
  rank?: number;
  feedback?: Feedback[];
  audience_votes: number;
  is_finalist: boolean;
  is_winner: boolean;
  created_at: string;
  updated_at: string;
  venture?: StudentVenture;
}

export interface SupportingDocument {
  name: string;
  url: string;
  type: string;
}

export interface JudgeScore {
  judge_id: number;
  judge_name: string;
  criteria_scores: CriteriaScore[];
  total_score: number;
  comments?: string;
}

export interface CriteriaScore {
  criteria_id: string;
  score: number;
}

export interface Feedback {
  judge_id: number;
  judge_name: string;
  comment: string;
  created_at: string;
}

export interface EntrepreneurshipMentor {
  id: number;
  institution_id?: number;
  user_id?: number;
  first_name: string;
  last_name: string;
  email: string;
  phone?: string;
  expertise_areas: string[];
  industry_experience?: IndustryExperience[];
  current_position?: string;
  company?: string;
  bio?: string;
  linkedin_url?: string;
  photo_url?: string;
  years_of_experience?: number;
  successful_ventures?: SuccessfulVenture[];
  mentoring_capacity: number;
  current_mentees: number;
  available_for_mentoring: boolean;
  preferred_communication?: string[];
  availability_schedule?: AvailabilitySlot[];
  total_mentorships: number;
  average_rating?: number;
  is_active: boolean;
  is_verified: boolean;
  verified_at?: string;
  created_at: string;
  updated_at: string;
}

export interface IndustryExperience {
  industry: string;
  years: number;
  role?: string;
}

export interface SuccessfulVenture {
  name: string;
  description: string;
  outcome: string;
  year?: number;
}

export interface AvailabilitySlot {
  day: string;
  start_time: string;
  end_time: string;
}

export interface MentorshipRelationship {
  id: number;
  institution_id: number;
  mentor_id: number;
  venture_id: number;
  match_score?: number;
  matching_criteria?: MatchingCriteria;
  status: MentorshipStatus;
  start_date?: string;
  end_date?: string;
  duration_weeks?: number;
  goals?: string[];
  meeting_frequency?: string;
  total_meetings: number;
  progress_notes?: ProgressNote[];
  student_feedback?: string;
  mentor_feedback?: string;
  student_rating?: number;
  mentor_rating?: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  mentor?: EntrepreneurshipMentor;
  venture?: StudentVenture;
}

export interface MatchingCriteria {
  expertise_match: number;
  industry_match: number;
  availability_match: number;
}

export interface ProgressNote {
  date: string;
  note: string;
  author: string;
}

export interface VentureFundingRequest {
  id: number;
  institution_id: number;
  venture_id: number;
  amount_requested: number;
  currency: string;
  funding_purpose: string;
  use_of_funds_breakdown?: FundsBreakdown[];
  justification: string;
  expected_outcomes?: string;
  supporting_documents?: SupportingDocument[];
  financial_projections?: FinancialProjection[];
  status: FundingStatus;
  reviewed_by?: number;
  review_date?: string;
  review_notes?: string;
  approved_amount?: number;
  disbursed_amount?: number;
  disbursement_date?: string;
  terms_and_conditions?: TermsCondition[];
  reporting_requirements?: ReportingRequirement[];
  created_at: string;
  updated_at: string;
  venture?: StudentVenture;
  reviewer?: UserInfo;
}

export interface FundsBreakdown {
  category: string;
  amount: number;
  description: string;
}

export interface FinancialProjection {
  period: string;
  revenue: number;
  expenses: number;
  profit: number;
}

export interface TermsCondition {
  title: string;
  description: string;
}

export interface ReportingRequirement {
  title: string;
  description: string;
  due_date: string;
}

export interface UserInfo {
  id: number;
  first_name: string;
  last_name: string;
  email: string;
}

export interface VentureBuilderTemplate {
  id: string;
  section: string;
  title: string;
  description: string;
  prompts: string[];
  examples?: string[];
}

export interface BusinessPlanData {
  executive_summary?: string;
  business_description?: string;
  market_analysis?: string;
  organization_management?: string;
  marketing_sales?: string;
  financial_projections?: FinancialProjection[];
  funding_request?: string;
  appendix?: string;
}

export interface PitchDeckSlide {
  id: string;
  type: string;
  title: string;
  content: string;
  layout: string;
  design?: SlideDesign;
}

export interface SlideDesign {
  background_color?: string;
  text_color?: string;
  font_family?: string;
  template?: string;
}

export interface LogoDesign {
  business_name: string;
  tagline?: string;
  color_scheme: string[];
  style: string;
  icon?: string;
  font_family?: string;
}

File: src/schemas/entrepreneurship.py
from datetime import datetime, date
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, EmailStr, Field, ConfigDict
from decimal import Decimal


class StudentVentureBase(BaseModel):
    venture_name: str = Field(..., max_length=200)
    tagline: Optional[str] = Field(None, max_length=300)
    founder_students: List[int]
    business_idea: str
    problem_statement: str
    solution: str
    target_market: str
    revenue_model: Optional[str] = None
    pitch_deck_url: Optional[str] = None
    business_plan_url: Optional[str] = None
    logo_url: Optional[str] = None
    website_url: Optional[str] = None
    funding_requested: Optional[Decimal] = None
    currency: str = "USD"
    social_impact: Optional[str] = None
    metrics: Optional[Dict[str, Any]] = None
    milestones: Optional[List[Dict[str, Any]]] = None
    team_roles: Optional[Dict[str, str]] = None


class StudentVentureCreate(StudentVentureBase):
    institution_id: int
    primary_founder_id: int


class StudentVentureUpdate(BaseModel):
    venture_name: Optional[str] = Field(None, max_length=200)
    tagline: Optional[str] = Field(None, max_length=300)
    business_idea: Optional[str] = None
    problem_statement: Optional[str] = None
    solution: Optional[str] = None
    target_market: Optional[str] = None
    revenue_model: Optional[str] = None
    pitch_deck_url: Optional[str] = None
    business_plan_url: Optional[str] = None
    logo_url: Optional[str] = None
    website_url: Optional[str] = None
    funding_requested: Optional[Decimal] = None
    venture_status: Optional[str] = None
    social_impact: Optional[str] = None
    metrics: Optional[Dict[str, Any]] = None
    milestones: Optional[List[Dict[str, Any]]] = None
    team_roles: Optional[Dict[str, str]] = None
    customers: Optional[int] = None
    revenue: Optional[Decimal] = None


class StudentVentureResponse(StudentVentureBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: int
    institution_id: int
    primary_founder_id: int
    mentor_id: Optional[int] = None
    funding_received: Decimal
    venture_status: str
    customers: int
    revenue: Decimal
    achievements: Optional[List[Dict[str, Any]]] = None
    pitch_competition_participations: Optional[List[int]] = None
    awards: Optional[List[Dict[str, Any]]] = None
    is_active: bool
    is_featured: bool
    created_at: datetime
    updated_at: datetime


class PitchCompetitionBase(BaseModel):
    competition_name: str = Field(..., max_length=200)
    description: str
    theme: Optional[str] = Field(None, max_length=200)
    judges: Optional[List[int]] = None
    judge_details: Optional[List[Dict[str, Any]]] = None
    prize_pool: Optional[Decimal] = None
    currency: str = "USD"
    prizes: Optional[List[Dict[str, Any]]] = None
    submission_deadline: datetime
    competition_date: Optional[datetime] = None
    eligibility_criteria: Optional[Dict[str, Any]] = None
    evaluation_criteria: Optional[List[Dict[str, Any]]] = None
    max_participants: Optional[int] = None


class PitchCompetitionCreate(PitchCompetitionBase):
    institution_id: int


class PitchCompetitionUpdate(BaseModel):
    competition_name: Optional[str] = Field(None, max_length=200)
    description: Optional[str] = None
    theme: Optional[str] = Field(None, max_length=200)
    judges: Optional[List[int]] = None
    judge_details: Optional[List[Dict[str, Any]]] = None
    prize_pool: Optional[Decimal] = None
    prizes: Optional[List[Dict[str, Any]]] = None
    submission_deadline: Optional[datetime] = None
    competition_date: Optional[datetime] = None
    eligibility_criteria: Optional[Dict[str, Any]] = None
    evaluation_criteria: Optional[List[Dict[str, Any]]] = None
    max_participants: Optional[int] = None
    status: Optional[str] = None
    winner_venture_id: Optional[int] = None
    runners_up: Optional[List[int]] = None
    leaderboard: Optional[List[Dict[str, Any]]] = None
    final_results: Optional[Dict[str, Any]] = None
    recording_url: Optional[str] = None
    highlights_url: Optional[str] = None
    is_public: Optional[bool] = None


class PitchCompetitionResponse(PitchCompetitionBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: int
    institution_id: int
    current_participants: int
    status: str
    winner_venture_id: Optional[int] = None
    runners_up: Optional[List[int]] = None
    leaderboard: Optional[List[Dict[str, Any]]] = None
    final_results: Optional[Dict[str, Any]] = None
    recording_url: Optional[str] = None
    highlights_url: Optional[str] = None
    is_public: bool
    is_active: bool
    created_at: datetime
    updated_at: datetime


class PitchSubmissionBase(BaseModel):
    pitch_video_url: Optional[str] = None
    presentation_url: Optional[str] = None
    supporting_documents: Optional[List[Dict[str, Any]]] = None


class PitchSubmissionCreate(PitchSubmissionBase):
    institution_id: int
    competition_id: int
    venture_id: int


class PitchSubmissionUpdate(BaseModel):
    pitch_video_url: Optional[str] = None
    presentation_url: Optional[str] = None
    supporting_documents: Optional[List[Dict[str, Any]]] = None


class PitchSubmissionResponse(PitchSubmissionBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: int
    institution_id: int
    competition_id: int
    venture_id: int
    submission_date: datetime
    judge_scores: Optional[Dict[str, Any]] = None
    total_score: Optional[Decimal] = None
    rank: Optional[int] = None
    feedback: Optional[List[Dict[str, Any]]] = None
    audience_votes: int
    is_finalist: bool
    is_winner: bool
    created_at: datetime
    updated_at: datetime


class EntrepreneurshipMentorBase(BaseModel):
    first_name: str = Field(..., max_length=100)
    last_name: str = Field(..., max_length=100)
    email: EmailStr
    phone: Optional[str] = Field(None, max_length=20)
    expertise_areas: List[Dict[str, Any]]
    industry_experience: Optional[List[Dict[str, Any]]] = None
    current_position: Optional[str] = Field(None, max_length=200)
    company: Optional[str] = Field(None, max_length=200)
    bio: Optional[str] = None
    linkedin_url: Optional[str] = None
    photo_url: Optional[str] = None
    years_of_experience: Optional[int] = None
    successful_ventures: Optional[List[Dict[str, Any]]] = None
    mentoring_capacity: int = 5
    preferred_communication: Optional[List[str]] = None
    availability_schedule: Optional[Dict[str, Any]] = None


class EntrepreneurshipMentorCreate(EntrepreneurshipMentorBase):
    institution_id: Optional[int] = None
    user_id: Optional[int] = None


class EntrepreneurshipMentorUpdate(BaseModel):
    first_name: Optional[str] = Field(None, max_length=100)
    last_name: Optional[str] = Field(None, max_length=100)
    email: Optional[EmailStr] = None
    phone: Optional[str] = Field(None, max_length=20)
    expertise_areas: Optional[List[Dict[str, Any]]] = None
    industry_experience: Optional[List[Dict[str, Any]]] = None
    current_position: Optional[str] = Field(None, max_length=200)
    company: Optional[str] = Field(None, max_length=200)
    bio: Optional[str] = None
    linkedin_url: Optional[str] = None
    photo_url: Optional[str] = None
    years_of_experience: Optional[int] = None
    successful_ventures: Optional[List[Dict[str, Any]]] = None
    mentoring_capacity: Optional[int] = None
    preferred_communication: Optional[List[str]] = None
    availability_schedule: Optional[Dict[str, Any]] = None
    available_for_mentoring: Optional[bool] = None


class EntrepreneurshipMentorResponse(EntrepreneurshipMentorBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: int
    institution_id: Optional[int] = None
    user_id: Optional[int] = None
    current_mentees: int
    available_for_mentoring: bool
    total_mentorships: int
    average_rating: Optional[Decimal] = None
    is_active: bool
    is_verified: bool
    verified_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime


class MentorshipRelationshipBase(BaseModel):
    goals: Optional[List[str]] = None
    meeting_frequency: Optional[str] = Field(None, max_length=50)


class MentorshipRelationshipCreate(MentorshipRelationshipBase):
    institution_id: int
    mentor_id: int
    venture_id: int


class MentorshipRelationshipUpdate(BaseModel):
    status: Optional[str] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    duration_weeks: Optional[int] = None
    goals: Optional[List[str]] = None
    meeting_frequency: Optional[str] = None
    total_meetings: Optional[int] = None
    progress_notes: Optional[List[Dict[str, Any]]] = None
    student_feedback: Optional[str] = None
    mentor_feedback: Optional[str] = None
    student_rating: Optional[int] = Field(None, ge=1, le=5)
    mentor_rating: Optional[int] = Field(None, ge=1, le=5)


class MentorshipRelationshipResponse(MentorshipRelationshipBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: int
    institution_id: int
    mentor_id: int
    venture_id: int
    match_score: Optional[Decimal] = None
    matching_criteria: Optional[Dict[str, Any]] = None
    status: str
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    duration_weeks: Optional[int] = None
    total_meetings: int
    progress_notes: Optional[List[Dict[str, Any]]] = None
    student_feedback: Optional[str] = None
    mentor_feedback: Optional[str] = None
    student_rating: Optional[int] = None
    mentor_rating: Optional[int] = None
    is_active: bool
    created_at: datetime
    updated_at: datetime


class VentureFundingRequestBase(BaseModel):
    amount_requested: Decimal
    currency: str = "USD"
    funding_purpose: str
    use_of_funds_breakdown: Optional[Dict[str, Decimal]] = None
    justification: str
    expected_outcomes: Optional[str] = None
    supporting_documents: Optional[List[Dict[str, Any]]] = None
    financial_projections: Optional[Dict[str, Any]] = None


class VentureFundingRequestCreate(VentureFundingRequestBase):
    institution_id: int
    venture_id: int


class VentureFundingRequestUpdate(BaseModel):
    status: Optional[str] = None
    review_notes: Optional[str] = None
    approved_amount: Optional[Decimal] = None
    disbursed_amount: Optional[Decimal] = None
    disbursement_date: Optional[date] = None
    terms_and_conditions: Optional[Dict[str, Any]] = None
    reporting_requirements: Optional[List[str]] = None


class VentureFundingRequestResponse(VentureFundingRequestBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: int
    institution_id: int
    venture_id: int
    status: str
    reviewed_by: Optional[int] = None
    review_date: Optional[datetime] = None
    review_notes: Optional[str] = None
    approved_amount: Optional[Decimal] = None
    disbursed_amount: Optional[Decimal] = None
    disbursement_date: Optional[date] = None
    terms_and_conditions: Optional[Dict[str, Any]] = None
    reporting_requirements: Optional[List[str]] = None
    created_at: datetime
    updated_at: datetime


class MentorMatchRequest(BaseModel):
    venture_id: int
    preferred_expertise: Optional[List[str]] = None
    preferred_industry: Optional[str] = None


class SubmitPitchRequest(BaseModel):
    competition_id: int
    venture_id: int
    pitch_video_url: Optional[str] = None
    presentation_url: Optional[str] = None
    supporting_documents: Optional[List[Dict[str, Any]]] = None


class JudgeScoreRequest(BaseModel):
    submission_id: int
    judge_id: int
    scores: Dict[str, Decimal]
    feedback: Optional[str] = None


class VentureShowcaseFilter(BaseModel):
    status: Optional[str] = None
    has_mentor: Optional[bool] = None
    min_funding: Optional[Decimal] = None
    is_featured: Optional[bool] = None
    search_query: Optional[str] = None
